src/main/java/org/kiwiproject/eureka/EmbeddedEurekaServer.java

```diff
@@ -0,0 +1,105 @@
+package org.kiwiproject.eureka;
+
+import static com.google.common.base.Preconditions.checkState;
+
+import lombok.extern.slf4j.Slf4j;
+import org.eclipse.jetty.server.Connector;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.ServerConnector;
+import org.eclipse.jetty.webapp.WebAppContext;
+import org.kiwiproject.io.KiwiPaths;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Creates, configures, starts and stops an embedded Jetty server running a Eureka Server.
+ * <p>
+ * If using this class outside of {@link org.kiwiproject.eureka.junit.EurekaServerExtension} then you may want to
+ * set the eureka.numberRegistrySyncRetries to zero in the System properties. This will allow the server to spin up
+ * quickly an not try to connect to other non-existent Eureka servers.
+ * See https://github.com/Netflix/eureka/issues/42#issuecomment-75614903 for dialog on this.
+ */
+@Slf4j
+public class EmbeddedEurekaServer {
+
+    private static final long IDLE_TIMEOUT = TimeUnit.HOURS.toMillis(1L);
+
+    private final Server eurekaServer;
+    private final ServerConnector connector;
+
+    /**
+     * Creates a new EmbeddedEurekaServer allowing Jetty to pick an available port.
+     */
+    public EmbeddedEurekaServer() {
+        this(0);
+    }
+
+    /**
+     * Creates a new EmbeddedEurekaServer with the given port.
+     *
+     * @param port the port to run Jetty on.
+     */
+    public EmbeddedEurekaServer(int port) {
+        eurekaServer = new Server();
+        connector = new ServerConnector(eurekaServer);
+
+        // Set timeout option to make debugging easier.
+        connector.setIdleTimeout(IDLE_TIMEOUT);
+
+        connector.setPort(port);
+        eurekaServer.setConnectors(new Connector[] { connector });
+
+        var context = new WebAppContext();
+        context.setServer(eurekaServer);
+        context.setContextPath("/");
+        context.setWar(KiwiPaths.pathFromResourceName("eureka-server.war").toString());
```

Comment:
Is there a way we can NOT have this giant WAR file in our repository? Like how the embedded Postgres and Mongo extensions download on-the-fly into a cached but well-known location? That way you incur the download on the first time, but not after that until and unless that cached file is deleted (or maybe we update to a newer version in which case we should cleanup the old version and download the newer version).